src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -599,5 +583,88 @@ private static string GetReferencedModel(String schema, Dictionary<string, Schem
 
         public static IEnumerable<string> GetParentTrackedResources(IEnumerable<string> trackedResourceModels, IEnumerable<KeyValuePair<string, string>> childTrackedResourceModels)
             => trackedResourceModels.Where(resourceModel => !childTrackedResourceModels.Any(childModel => childModel.Key.Equals(resourceModel)));
+
+        /// <summary>
+        /// For the provided resource model, it gets the operation which ends with ListByResourceGroup and returns the resource model.
+        /// </summary>
+        /// <param name="resourceModel"></param>
+        /// <param name="definitions"></param>
+        /// <param name="serviceDefinition"></param>
+        /// <returns></returns>
+        public static Operation GetListByResourceGroupOperation(string resourceModel, Dictionary<string, Schema> definitions, ServiceDefinition serviceDefinition)
```

Comment:
Starting to wonder if we can create a separate class for listBy related methods and reduce some of the load in ValidationUtilities.